eeco/utils.py

```diff
@@ -312,21 +310,238 @@ def const_rule(model):
-        model.add_component(varstr + "_constraint", constraint)
-        return (var, model)
+            constraint = pyo.Constraint(expression.index_set(), rule=const_rule)
+            model.add_component(varstr + "_constraint", constraint)
+            return (var, model)
+        else:
+            # Create scalar max_pos variable
+            model.add_component(varstr, pyo.Var(initialize=0, bounds=(0, None)))
+            var = model.find_component(varstr)
+
+            def const_rule(model):
+                return var >= expression
+
+            constraint = pyo.Constraint(rule=const_rule)
+            model.add_component(varstr + "_constraint", constraint)
+            return (var, model)
     elif isinstance(
         expression, (int, float, np.int32, np.int64, np.float32, np.float64, np.ndarray)
     ):
         return (np.max(expression), model) if np.max(expression) > 0 else (0, model)
     elif isinstance(expression, cp.Expression):
-        return cp.max(cp.vstack([expression, 0])), None
+        # Check if expression is indexed
+        if expression.shape == ():  # Scalar
+            return cp.maximum(expression, 0), None
+        else:  # Vector
+            return cp.maximum(expression, 0), None
+    else:
+        raise TypeError(
+            "Only CVXPY or Pyomo variables and NumPy arrays are currently supported."
+        )
+
+
+def _get_decomposed_var_names(utility):
+    """Get consistent variable names for decomposed consumption variables."""
+    return f"{utility}_positive", f"{utility}_negative"
+
+
+def initialize_decomposed_pyo_vars(consumption_data_dict, model, charge_dict):
+    """Helper function to initialize Pyomo variables with baseline consumption values.
+
+    This function takes consumption data as numpy arrays, decomposes them using
+    the numpy version of decompose_consumption, and then initializes the corresponding
+    Pyomo variables with those values.
+
+    Parameters
+    ----------
+    consumption_data_dict : dict
+        Dictionary with keys "electric" and "gas" containing numpy arrays
+        of consumption data.
+
+    model : pyomo.environ.Model
+        The Pyomo model containing the variables to initialize.
+
+    charge_dict : dict
+        Dictionary containing charge arrays for different utilities and charge types.
+        Used to extract the correct charge rate for export calculations.
+
+    Returns
+    -------
+    dict
+        Dictionary with initialized consumption objects for each utility.
+    """
+    consumption_object_dict = {}
+
+    # Initialize the basic consumption variables and converted variables
+    for utility in consumption_data_dict.keys():
+        consumption_data = consumption_data_dict[utility]
+        consumption_var = model.find_component(f"{utility}_consumption")
+        if consumption_var is not None:
+            for t in model.t:
+                consumption_var[t].value = consumption_data[t - 1]  # Pyomo 1-indexed
```

Comment:
Our other functions use `model.t = range(model.T)` which starts at 0, so I do not think that we should assume that it will be 1-based indexing.

Reply:
I forgot to ask if you checked this with `model.t` starting at 0. The most robust solution might be to check model.t[0] and if it is zero then do nothing but if it is 1 then translate from 1-based to 0-based indexing